File: NaviTest/NaviTest/PinkieNavigation/NavigationController/PinkieNavigationController.m
//
//  PinkieNavigationController.m
//  NaviTest
//
//  Created by jiazhiyi on 2019/4/9.
//  Copyright © 2019 jiazhiyi. All rights reserved.
//

#import "PinkieNavigationController.h"
#import "PinkiePopAnimatedTransitioning.h"
#import "PinkieNavigationDelegateImp.h"

@interface PinkieNavigationController ()

@property (nonatomic, strong) PinkiePopInteractiveTransitioning     *popTransition;
@property (nonatomic, strong) PinkieNavigationDelegateImp           *delegateImp;

@end

@implementation PinkieNavigationController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.navigationBarHidden = YES;
    [self setupDelegateImp];
    [self setupUI];
}

- (void)setupDelegateImp
{
    self.popTransition = [[PinkiePopInteractiveTransitioning alloc] init];
    self.popTransition.navigationController = self;
    
    self.delegateImp = [[PinkieNavigationDelegateImp alloc] init];
    self.delegateImp.transitioning = self.popTransition;
    self.delegate = self.delegateImp;
    self.interactivePopGestureRecognizer.enabled = NO;
}

- (void)setupUI
{
    [self setupNavigationView];
}

- (void)setupNavigationView
{
    CGFloat height = 64.0;
    CGFloat width = self.view.frame.size.width;
    CGRect rect = CGRectMake(0.0, 0.0, width, height);
    self.navigationView = [[PinkieNavigationView alloc] initWithFrame:rect];
    self.navigationView.backgroundColor = [UIColor redColor];
    [self.view addSubview:self.navigationView];
}


@end

File: CustomNavigationBar/CustomNavigationBar/PinkieNavigation/NavigationController/PinkieNavigationController.swift
//
//  PinkieNavigationController.swift
//  CustomNavigationBar
//
//  Created by jiazhiyi on 2019/4/4.
//  Copyright © 2019 jiazhiyi. All rights reserved.
//

import UIKit

class PinkieNavigationController: UINavigationController {
    
    var navigationView: PinkieNavigationView?
    let popTransition = PinkieNavPopTransitioning()
    var delegateImp: PinkieNavigationDelegateImp?

    override func viewDidLoad() {
        super.viewDidLoad()

        self.isNavigationBarHidden = true
        
        self.setupDelegateImp()
        self.setupUI()
    }
    
    func setupDelegateImp() {
        self.popTransition.navigationController = self
        
        self.delegateImp = PinkieNavigationDelegateImp(popTransition: self.popTransition)
        self.delegate = self.delegateImp
        self.interactivePopGestureRecognizer?.isEnabled = false
    }
    
    func setupUI() {
        self.setupNavigationView()
    }

    
    func setupNavigationView() {
        let height = 64.0
        let width = self.view.frame.width
        let rect = CGRect(x: 0.0, y: 0.0, width: Double(width), height: height)
        self.navigationView = PinkieNavigationView(frame: rect)
        self.navigationView?.backgroundColor = UIColor.red
        self.view.addSubview(self.navigationView!)
    }
}
